lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -3216,6 +3216,10 @@ ConstraintResult GenericSignatureBuilder::expandConformanceRequirement(
     }
   }
 
+  // Remaining logic is not relevant in imported-from-ObjC protocol cases.
+  if (proto->isObjC() && proto->hasClangNode())
+    return ConstraintResult::Resolved;
```

Comment:
Why is this specific to imported-from-ObjC protocols and not all `@objc` protocols?

Reply:
Well .. there was a testcase that seems to care (compiler crasher 28855-swift-genericsignaturebuilder...) but in general I think this code path is about improving diagnostics in cases that might credibly occur (according to Slava at least) and the cases we're cutting off are those where there's a typealias in a protocol. And that's a fair bit easier to write in swift as `@objc protocol P { typealias ... ` (as in the testcase) than it is likely to arrive through a clang-import-as-member (which Slava at least implied was not worth worrying about here; of course your opinion may vary!)

Reply:
Ok if it regresses a crasher, sure. Maybe @DougGregor has ideas.

Reply:
I feel like whatever special things the GSB does with type aliases, it should not do them with objc protocols. Treat them like any other nested (nominal) type.

Reply:
I'm happy to just rewrite the crasher too, if this fix makes it no-longer-crash. I was just being a bit lazy. Preference?